awswrangler/distributed/ray/datasources/pandas_file_based_datasource.py

```diff
@@ -1,10 +1,12 @@
 """Ray PandasFileBasedDatasource Module."""
 import logging
-from typing import Any, Callable, Dict, List, Optional
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Iterable, List, Optional
 
 import pandas as pd
 import pyarrow
-from ray.data.block import Block, BlockAccessor, BlockMetadata
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
```

Comment:
Would these changes be backward compatible? Just wary of accessing private modules which are more likely to break...

Reply:
No they would not, as I understand this was added to the PR just to make it work it with ray 2.4. 

It looks like 2.4 had a number of backward-incompatible changes such as [deprecation of do_write](https://github.com/ray-project/ray/pull/32015/).

Reply:
Reg. this specific `DelegatingBlockBuilder` class is available in all 2.x version of Ray. It's a shame it is not exported but I think we should be fine. This and `OutputBlockBuffer` are the only internal Ray classes we depend on and both are quite key to ray I would be surprised if they change the implementation drastically or move. Both situations should be caught by our tests.